src/cli.rs

```diff
@@ -82,7 +83,22 @@ pub async fn cli() -> Result<(), Box<dyn Error>> {
     rl.set_helper(Some(h));
     let _ = rl.load_history("history.txt");
 
+    let ctrl_c = Arc::new(AtomicBool::new(false));
+    let cc = ctrl_c.clone();
+    ctrlc::set_handler(move || {
+        cc.store(true, Ordering::SeqCst);
+    })
+    .expect("Error setting Ctrl-C handler");
+
     loop {
+        if ctrl_c.load(Ordering::SeqCst) {
+            ctrl_c.store(false, Ordering::SeqCst);
+            if let ShellError::String(s) = ShellError::string("CTRL-C") {
+                context.host.lock().unwrap().stdout(&format!("{:?}", s));
+            }
```

Comment:
This feels a bit messy repeating the error, but I think it's best to return the same error type so Ctrl-C behaves the same when terminating a subprocess.

I couldn't create a `StringError` directly because `Value` is private, so I'm pattern matching to deconstruct the `ShellError`.